src/charm.py

```diff
@@ -530,6 +550,91 @@ def _on_new_vault_kv_client_attached(self, event: NewVaultKvClientAttachedEvent)
         stale_nonces = set(credential_nonces) - set(nonces)
         self.vault_kv.remove_unit_credentials(relation, stale_nonces)
 
+    def _on_certificates_pki_request_relation_joined(self, event: RelationJoinedEvent):
+        """Initializes the PKI backend."""
+        if not self.unit.is_leader():
+            logger.debug("Only leader unit should configure the vault PKI, skipping")
+            return
+        if not self._is_peer_relation_created():
+            logger.debug("Peer relation not created, deferring event")
+            event.defer()
+            return
+        common_name = self._get_config_common_name()
+        if not common_name:
+            logger.error("Common name is not set correctly in the charm configuration")
+            return
```

Comment:
Seems like we need to handle the configuration change, otherwise the charm will get stuck.